packages/mdc-textfield/foundation.ts

```diff
@@ -22,127 +22,110 @@
   }
 
   /**
-   * {@see MDCTextFieldAdapter} for typing information on parameters and return
-   * types.
-   * @return {!MDCTextFieldAdapter}
+   * See {@link MDCTextFieldAdapter} for typing information on parameters and return types.
    */
-  static get defaultAdapter() {
-    return /** @type {!MDCTextFieldAdapter} */ ({
-      addClass: () => {},
-      removeClass: () => {},
-      hasClass: () => {},
-      registerTextFieldInteractionHandler: () => {},
-      deregisterTextFieldInteractionHandler: () => {},
-      registerInputInteractionHandler: () => {},
-      deregisterInputInteractionHandler: () => {},
-      registerValidationAttributeChangeHandler: () => {},
-      deregisterValidationAttributeChangeHandler: () => {},
-      getNativeInput: () => {},
-      isFocused: () => {},
-      activateLineRipple: () => {},
-      deactivateLineRipple: () => {},
-      setLineRippleTransformOrigin: () => {},
-      shakeLabel: () => {},
-      floatLabel: () => {},
-      hasLabel: () => {},
-      getLabelWidth: () => {},
-      hasOutline: () => {},
-      notchOutline: () => {},
-      closeOutline: () => {},
-    });
-  }
+  static get defaultAdapter(): MDCTextFieldAdapter {
+    // tslint:disable:object-literal-sort-keys
+    return {
+      addClass: () => undefined,
+      removeClass: () => undefined,
+      hasClass: () => true,
+      registerTextFieldInteractionHandler: () => undefined,
+      deregisterTextFieldInteractionHandler: () => undefined,
+      registerInputInteractionHandler: () => undefined,
+      deregisterInputInteractionHandler: () => undefined,
+      registerValidationAttributeChangeHandler: () => new MutationObserver(() => undefined),
+      deregisterValidationAttributeChangeHandler: () => undefined,
+      getNativeInput: () => null,
+      isFocused: () => false,
+      activateLineRipple: () => undefined,
+      deactivateLineRipple: () => undefined,
+      setLineRippleTransformOrigin: () => undefined,
+      shakeLabel: () => undefined,
+      floatLabel: () => undefined,
+      hasLabel: () => false,
+      getLabelWidth: () => 0,
+      hasOutline: () => false,
+      notchOutline: () => undefined,
+      closeOutline: () => undefined,
+    };
+    // tslint:enable:object-literal-sort-keys
+  }
+
+  private isFocused_ = false;
+  private receivedUserInput_ = false;
+  private isValid_ = true;
+  private useNativeValidation_ = true;
+
+  private readonly inputFocusHandler_: () => void;
+  private readonly inputBlurHandler_: EventListener;
+  private readonly inputInputHandler_: EventListener;
+  private readonly setPointerXOffset_: EventListener;
+  private readonly textFieldInteractionHandler_: EventListener;
+  private readonly validationAttributeChangeHandler_: (attributesList: string[]) => void;
+  private validationObserver_!: MutationObserver; // assigned in init()
+
+  private readonly helperText_: MDCTextFieldHelperTextFoundation | undefined;
```

Comment:
change the `| undefined` --> `helperText_?: MDCTextFieldHelperTextFoundation`, as well as the rest below

Reply:
Done